common/script/index.coffee

```diff
@@ -646,6 +646,20 @@ api.wrap = (user, main=true) ->
         cb? null, user.preferences.webhooks
 
       # ------
+      # Push Notifications
+      # ------
+      addPushDevice: (req, cb) ->
+        user.pushDevices = [] unless user.pushDevices
+        pd = user.pushDevices
+        item = {regId:req.body.regId, type:req.body.type};
+        i = _.findIndex pd, {regId: item.regId}
+
+        pd.push(item) unless i != -1
+
+        user.markModified? 'pushDevices'
```

Comment:
you can remove the markModfied statement, in addWebhook it's used because it's not really an array but a `Mixed` schema that requires it